### DISCUSSION TITLE
Deploying to Netlify, previous querystring gets appended to the redirect

### DISCUSSION THREAD

**Discussion by robations** [sveltejs/kit#11345] [Q&A]
This seems to work as expected on the local dev server, but when the app is deployed to Netlify, any query string parameter (such as `?foo=bar`) gets appended to the redirect URL.

The URL, which contains a form:

https://app.example.com/?foo=bar

POSTing the form to itself (`action=""`) will then redirect to a new URL:

```ts
throw redirect(301, "https://external.example.com/");
```

Actual outcome:

```
Location: https://external.example.com/?foo=bar
```

All attempts to prevent this happening (different 30x codes, using a string URL, using a URL object) have failed me so far. Testing requires a commit/push/rebuild loop, so it's very slow to try different options.

Is there a known workaround, is it a bug/feature in Netlify/Lambda, would it be a feature request for sveltekit or the Netlify adaptor?

Actually I have a workaround, which is to set the post action to `"?"` and `target="_blank"`, but I'd be interested if anyone has any thoughts.

**Comment by duylaurent-tran** [comment]
Same issues here. I am still looking for a solution.

**Comment by robations** [comment]
Seems like I'm not the only one. Can anyone suggest the appropriate project to raise this as a bug?
